pennylane/devices/qubit/simulate.py

```diff
@@ -124,24 +133,31 @@ def get_final_state(circuit, debugger=None, interface=None):
 
     # initial state is batched only if the state preparation (if it exists) is batched
     is_state_batched = bool(prep and prep.batch_size is not None)
+    measurement_values = {}
     for op in circuit.operations[bool(prep) :]:
+        if isinstance(op, Conditional):
+            meas_id = op.meas_val.measurements[0].hash
```

Comment:
Not necessary for this PR but it might be worth adding a `hash` property to `MeasurementValue` that depends on the hash of its `measurements`. Something like `self.hash = hash(tuple(self.measurements))`. Could be worth trying out in the follow up where multi-measurement conditioning support is added.

Reply:
Good point. I think adding this `hash` might be necessary, or at least the path of least resistance. Will do in the follow-up PR as suggested.